src/vm/docs/mod.rs

```diff
@@ -264,6 +264,46 @@ Note: This function is only available starting with Stacks 2.1.",
 "#,
 };
 
+const PARSE_PRINCIPAL_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(parse-principal property-name principal-address)",
+    description: "A principal value is a concatenation of two things: a single *version byte*
+(indicating the type of account and the type of network that this principal can spend tokens on),
+and a 20-byte *public key hash* (indicating the principal's unique identity).
+`parse-principal` is used to decompose a principal into its component parts.
+
+By setting `property-name` to `version`, one recovers version byte.
+
+By setting `property-name` to `pub-key-hash`, one recovers the 20-byte buffer representing the 
+public key hash of the user.
+
+Note: This function is only available starting with Stacks 2.1.",
+    example: r#"
+(parse-principal version 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns u26
+(parse-principal pub-key-hash 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns 0x164247d6f2b425ac5771423ae6c80c754f7172b0
+"#,
+};
+
+const ASSEMBLE_PRINCIPAL_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(assemble-principal version-byte public-key-hash)",
```

Comment:
I'm confused by the `version-byte` name. If it's a byte, it should be `(buff 1)`, not a uint?

Reply:
You think I should change the signature to accept `(buff 1)`? I can do it as long as we're sure and I don't have to change it back.

Reply:
Yeah I think we should change and use a `buff 1`, if possible.

Reply:
(for developer experience + cost optimization)